packages/realm/src/Realm.ts

```diff
@@ -1080,50 +1178,330 @@ function isEmbedded(objectSchema: binding.ObjectSchema): boolean {
+type SetType<T> = RealmSet<T>;
+type SyncErrorType = SyncError;
 type TypesType = typeof Types;
-type UserType = typeof User;
-type CredentialsType = typeof Credentials;
-type ConfigurationType = Configuration;
-type FlexibleSyncConfigurationType = FlexibleSyncConfiguration;
+type UpdateModeType = UpdateMode;
+type UserType = User;
+type UserStateType = UserState;
+
+type BaseSubscriptionSetType = BaseSubscriptionSet;
+type LogLevelType = LogLevel;
+type NumericLogLevelType = NumericLogLevel;
+type MutableSubscriptionSetType = MutableSubscriptionSet;
+type PartitionValueType = PartitionValue;
+type SubscriptionOptionsType = SubscriptionOptions;
+type SubscriptionSetType = SubscriptionSet;
+type SubscriptionsStateType = SubscriptionsState;
+type SubscriptionType = Subscription;
+type SyncSessionType = SyncSession;
+
+type ObjectIdType = BSON.ObjectId;
+type Decimal128Type = BSON.Decimal128;
+type UUIDType = BSON.UUID;
+
+type ApiKeyType = ApiKey;
+type EmailPasswordAuthType = EmailPasswordAuth;
+type ApiKeyAuthType = ApiKeyAuth;
+
+type GlobalDate = Date;
 
 // eslint-disable-next-line @typescript-eslint/no-namespace
-export namespace Realm {
+export declare namespace Realm {
+  // TODO: Decide if we want to deprecate this as well
   export type Object<T = DefaultObject> = RealmObject<T>;
-  export type Collection<T = unknown> = CollectionType<T>;
-  export type OrderedCollection<T = unknown> = OrderedCollectionType<T>;
-  export type Results<T = unknown> = ResultsType<T>;
-  export type List<T = unknown> = ListType<T>;
-  export type Dictionary<T = unknown> = DictionaryType<T>;
-  export type Set<T = unknown> = SetType<T>;
-  export type App = AppType;
-  export type UpdateMode = UpdateModeType;
-  export type ObjectSchema = ObjectSchemaType;
-  /**
-   * @deprecated Will be removed in v13.0.0. Please use {@link PropertySchema}.
-   */
-  export type ObjectSchemaProperty = ObjectSchemaPropertyType;
-  export type PropertySchema = PropertySchemaType;
-  export type PropertySchemaShorthand = PropertySchemaShorthandType;
-  export type Mixed = unknown;
-  export type BSON = BSONType;
-  export type Types = TypesType;
-  export type User = UserType;
-  export type Credentials = CredentialsType;
-  export type Configuration = ConfigurationType;
-  export type FlexibleSyncConfiguration = FlexibleSyncConfigurationType;
+  export {
+    // Pure type exports below
+    /** @deprecated Please use named imports */
+    AppType as App,
+    /** @deprecated Please use named imports */
+    AppChangeCallback,
+    /** @deprecated Please use named imports */
+    AppConfiguration,
+    /** @deprecated Please use named imports */
+    BaseConfiguration,
+    /** @deprecated Please use named imports */
+    BaseObjectSchema,
+    /** @deprecated Please use named imports */
+    BaseSyncConfiguration,
+    /** @deprecated Please use named imports */
+    BSONType as BSON,
+    /** @deprecated Please use named imports */
+    CanonicalObjectSchemaProperty,
```

Comment:
We can refer the user here to the now-called `CanonicalPropertySchema` (also needs to be imported at the top).
```suggestion
    /** @deprecated Will be removed in v13.0.0. Please use {@link CanonicalPropertySchema} as a named import */
    CanonicalObjectSchemaProperty,
    /** @deprecated Please use named imports */
    CanonicalPropertySchema,
```

Feel free to rearrange/reformulate.